rust/interpreter/src/value.rs

```diff
@@ -1,11 +1,16 @@
+use std::cell::RefCell;
 use std::f64;
 use std::str::FromStr;
+use std::rc::Rc;
+
+use crate::object::Object;
 
 #[derive(Clone, Debug)]
 pub enum JSValue {
     Boolean(bool),
     Number(f64),
     String(String),
+    Object(Rc<RefCell<Object>>),
```

Comment:
This will be a memory leak, of course, once objects exist. Boa uses the `gc` crate; other garbage collectors for Rust exist too, though I don't know that of any that aim to be "production-ready".

Reply:
Yeah, leaking is fine for now. Jonco had this idea of implementing a cycle-collector instead of a proper GC :)